utils/retry.js

```diff
@@ -1,6 +1,6 @@
 module.exports = function retry(fun, options = { timeout: 0, retries: 0 }) {
   return fun().catch((err) => {
-    if (options.retries === 0) {
+    if (options.retries <= 0 || options.timeout < 0) {
```

Comment:
ok, 

```
According to the MDN reference, the specification requires that there is a minimum timeout.

If you provide something less than this (HTML5 spec says 4ms) then the browser will just ignore your delay and use the minimum.

So negatives should be fine, since it'll just be less than the minimum.
```